packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -76,19 +76,30 @@ export interface CheckoutEvents {
 	/**
 	 * {@inheritdoc TreeViewEvents.commitApplied}
 	 */
-	commitApplied(data: CommitMetadata, getRevertible?: () => Revertible): void;
-
-	/**
-	 * Fired when a revertible is either reverted or discarded.
-	 *
-	 * This event can be used to maintain a list or set of active revertibles.
-	 * @param revertible - The revertible that was disposed.
-	 * This revertible was previously passed to the `newRevertible` event.
-	 * Calling `discard` on this revertible is not necessary but is safe to do.
-	 */
-	revertibleDisposed(revertible: Revertible): void;
+	commitApplied(data: CommitMetadata, getRevertible?: RevertibleFactory): void;
 }
 
+/**
+ * Factory for creating a {@link Revertible}.
+ *
+ * @param onRevertibleDisposed - A callback that will be invoked when the `Revertible` generated by this factory is disposed.
+ * Note that `SharedTree` retains the right to unilaterally dispose of revertibles. This happens when the view that
```

Comment:
This reads a little litigious. I'm okay with it if you prefer it, but I'd also suggest something like, "This happens when the Revertible is disposed manually, but may also happen for other reasons - for example, if the `TreeView` that the revertible belongs to is disposed."

Reply:
"Litigious" is my middle name 😎
I've tweaked your suggestion to be more specific, LMK what you think.

Reply:
Looks great!